docs/micro-service-architecture.md

```diff
@@ -0,0 +1,248 @@
+This service works closely with the Keycloak service.
+
+
+### `Keycloak`
+
+It will allow us to provide a secure authentication and authorization service for the client. 
+
+### `Notification Service`
+
+The Notification Service is responsible for sending notifications to the users. The goal is to have a single service responsible for sending notifications to the users.
+
+### `Job Service`
+
+The Job offer service is responsible for managing jobs, job offers and companies. This service is implemented in kotlin using spring r2dbc as a reactive database driver. 
+We chose r2dbc instead of jpa since being we needed to have asynchronous requests in our architecure to optimize latency. 
+Having a synchronous system would mean we can only treat one request at the time which at larger scale will not be efficient enough. 
+The service uses a postgres database to store it's data.
+
+### `Employer Service`
+
+The Employer Service represents the external software that employers uses to manage their job offers. 
+
+LinkedOut calls to this external software for retrieving job offers, or giving evaluations to employers for instance. 
+On the other hand, this external software calls to LinkedOut when an employer wants to message a seasonal worker, or to give an evaluation to a seasonal worker for instance.
+
+Since the employer service is normally an external service, we will only implement something that mocks it.
+
+### `Profile Service`
+
+The Profile Service is responsible for managing the profiles of the users. This service will be written in Kotlin as a matter of uniformity with the other services. 
+Profiles will be stored in a Postgres database and also uses r2dbc as it's database driver.
+
+### `NATS`
+
+NATS is a "Message-oriented Middleware": it allows services to communicate with each other easily. 
+
+NATS is the central element in our distributed and scalable infrastructure because it provides dynamic discovery and a message broker. 
+
+This will allow us to manage replica sets of services easily since NATS will abstract the concrete connection to the service using its 
+mechanism of "subjects" (which are like Kafka's topics) and will load balance messages between all consumers subscribed to a subject in particular. 
+
+### `Message Service`
+
+The message service is here to manage communications between seasonal workers and employers, allowing them to tchat using an instant messagery.
+This service will have persistent storage provided by a Postgresql database.
+ 
+### `Recommendation Service`
+
+The Recommendation Service is used to recommend job offers to seasonal workers. 
+It will have a read access to the job and profile services and replicate the data in a graph database (probably Neo4j) to find connections 
+between related items and provide scores on the job offers for a seasonal worker.
```

Comment:
```suggestion
The Recommendation Service is used to recommend job offers to seasonal workers. 

It will have read access to the job and profile services and replicate the data in a graph database (such as Neo4j) to find connections between related items and provide scores on the job offers for a seasonal worker.
```